ludwig/features/boundingbox_feature.py

```diff
@@ -0,0 +1,333 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import logging
+import os
+from collections import OrderedDict
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.ops.losses.losses_impl import Reduction
+
+from ludwig.constants import (BOUNDING_BOX,
+                              FILL_WITH_CONST,
+                              HUBER_LOSS,
+                              PREDICTIONS,
+                              PREDICTION,
+                              PROBABILITIES,
+                              ERROR,
+                              IOU,
+                              LOSS,
+                              EVAL_LOSS,
+                              TRAIN_MEAN_LOSS,
+                              MEASURE,
+                              SUM,
+                              APPEND)
+from ludwig.features.base_feature import BaseFeature
+from ludwig.features.base_feature import OutputFeature
+from ludwig.models.modules.initializer_modules import get_initializer
+from ludwig.models.modules.measure_modules import bbox_iou as get_bbox_iou
+from ludwig.models.modules.measure_modules import error as get_error
+from ludwig.utils.misc import set_default_value
+
+
+class BoundingBoxBaseFeature(BaseFeature):
+    def __init__(self, feature):
+        super().__init__(feature)
+        self.type = BOUNDING_BOX 
+
+    preprocessing_defaults = {
+        'missing_value_strategy': FILL_WITH_CONST,
+        'fill_value': 0
+    }
+
+    @staticmethod
+    def get_feature_meta(column, preprocessing_parameters):
+        return {}
+
+    @staticmethod
+    def add_feature_data(
+            feature,
+            dataset_df,
+            data,
+            metadata,
+            preprocessing_parameters,
+    ):
+        data[feature['name']] = np.stack(dataset_df[feature['name']], 
```

Comment:
How do you expect the content of the column containing the bounding box to look like? I imageine something like this:
```
image_path,bbox
path.jpeg,10 20 15 15
```
so the bbox column should contain 4 values divided by space.
In this case, I believe the processing should be different from what you are doing, but maybe you are thinking about another format.

Reply:
It is currently accepting bounding boxes as np arrays, but I agree that 4 values divided by space  is more convenient especially when dealing with files. I guess it means the bbox values would be handled as strings in python, right?